app/models/resource.rb

```diff
@@ -105,6 +105,14 @@ def search account: nil, kind: nil, owner: nil, offset: nil, limit: nil, search:
       scope = scope.textsearch(search) if search
 
       if offset || limit
+        # 'limit' must be an integer greater than 0 if given
+        if limit && (!numeric?(limit) || limit.to_i <= 0)
+          raise "'limit' contains an invalid value. 'limit' must be a positive integer."
+        end
+        # 'offset' must be an integer greater than or equal to 0 if given
+        if offset && (!numeric?(offset) || offset.to_i.negative?)
+          raise "'offset' contains an invalid value. 'offset' must be an integer greater than or equal to 0."
```

Comment:
@liavyona is it possible to move these errors to `errors.rb`?
we prefer to have all the user-facing messages there

Reply:
I think it will be a problem to use a custom exception inside a model class. Ads you can see there is no usage of any error which is defined at `errors.rb` under `app/model`. Other model classes use simple `raise` or `ArgumentError`. I changed it to use `ArgumentError`